src/v/rpc/connection_cache.cc

```diff
@@ -99,4 +110,21 @@ ss::future<> connection_cache::stop() {
     return _gate.close();
 }
 
+ss::shard_id connection_cache::shard_for(
+  model::node_id self,
+  ss::shard_id src_shard,
+  model::node_id n,
+  ss::shard_id total_shards) const {
+    if (total_shards <= _max_connections) {
+        return src_shard;
+    }
+
+    // NOLINTNEXTLINE
+    size_t h = 805306457;
```

Comment:
The `universe` thing where we select a prime depending on the source shard is gone. What was its purpose?

Reply:
It was to more uniformly distribute connections as far as I can tell. I need an analysis of this hash vs the one with a "universe" in the PR I linked. The connection distribution of both is pretty close. @mmaslankaprv may have more insight here as well.